bridge/slack/helpers.go

```diff
@@ -106,6 +107,65 @@ func (b *Bslack) populateChannels() {
 	b.channelsByName = newChannelsByName
 }
 
+// populateReceivedMessage shapes the initial Matterbridge message that we will forward to the
+// router before we apply message-dependent modifications.
+func (b *Bslack) populateReceivedMessage(ev *slack.MessageEvent) (*config.Message, error) {
+	// Use our own func because rtm.GetChannelInfo doesn't work for private channels.
+	channel, err := b.getChannelByID(ev.Channel)
+	if err != nil {
+		return nil, err
+	}
+
+	rmsg := &config.Message{
+		Text:    ev.Text,
+		Channel: channel.Name,
+		Account: b.Account,
+		ID:      "slack " + ev.Timestamp,
+		Extra:   make(map[string][]interface{}),
+	}
+	if b.useChannelID {
+		rmsg.Channel = "ID:" + channel.ID
+	}
+
+	if err = b.populateMessageWithUserInfo(ev, rmsg); err != nil {
+		return nil, err
+	}
+	return rmsg, err
+}
+
+func (b *Bslack) populateMessageWithUserInfo(ev *slack.MessageEvent, rmsg *config.Message) error {
+	if ev.SubType == sMessageDeleted || ev.SubType == sFileComment {
+		return nil
+	}
+
+	if ev.BotID != "" && b.GetString(tokenConfig) != "" {
```

Comment:
We should check `b.GetString(outgoingWebhookConfig) == ""` like in the original code, because people can (and do) combine outgoing webhook for sending messages and token for receiving.
Don't do any of this stuff when we have a `outgoingWebhookConfig` because we're not sure if it's our webhook or not

Reply:
What do you specifically mean here by "our webhook"?

----

May I ask what the reasons is for using webhooks for sending when already having an RTM connection available?

Context: In [my last follow-up commit](/Helcaraxan/matterbridge/commit/a7967527ceb436d082204e8d42884f849c65feb6) I was planning on simplifying the connection logic to use RTM for both sending and receiving when a token is available and only use webhooks if specified and if there is no token. This is also the reason why, maybe too early I flipped over the logic to check for the presence of the RTM token.

Reply:
Users want to have this option because with the webhooks they can "impersonate" a user, (changing the IconURL (avatar) and changing the username) with the RTM, you can't change your username or the icon. That's why we allow the use of both options (in a lot of other bridges too)